plugins/event_notification/providers/email/lib/batch/recipients/KEmailNotificationCategoryRecipientEngine.php

```diff
@@ -44,7 +44,25 @@ function getRecipients(array $contentParameters)
 			$userList = KBatchBase::$kClient->user->listAction($userFilter, $userPager);
 			foreach ($userList->objects as $user)
 			{
-				$recipients[$user->email] = $user->firstName. ' ' . $user->lastName;
+				if($user->type == KalturaUserType::USER)
+				{
+					$recipients[$user->email] = $user->firstName. ' ' . $user->lastName;
+				}
+				else if($user->type == KalturaUserType::GROUP)
+				{
+					$groupUserIds = $this->getGroupUserIds($user->id);
+					if(!$groupUserIds)
+						return $recipients;
```

Comment:
why at this point you stop and return the recipients? you should break from specific iteration, not fully return.
what if you have more users to run on in the for each loop, you wont get to them
same question for line 59